### PR TITLE
[telnetpp] Update port version 2.1.2 -> 3.1.0

### PR DESCRIPTION
**Pull request by kafeg**

### CODE REVIEW

File: ports/telnetpp/portfile.cmake [hunk 1]
```diff
@@ -1,10 +1,10 @@
 vcpkg_from_github(
   OUT_SOURCE_PATH SOURCE_PATH
   REPO KazDragon/telnetpp
-  REF f370ebd0c0dc6505708065ee5afdc59a6de54387 # v2.1.2 + MSVC patches
-  SHA512 c58cb9159a8fb6c4b089a0212a995f70f08b93877d98828aa263e9f065f42a932d98749b56741d9e711c0805dcc2dcf0607dc86b0553c4e34bd3fad99e0bf157
+  REF v${VERSION}
```
Comment (jimwang118) [inline ports/telnetpp/portfile.cmake:4]:
```suggestion
  REF "v${VERSION}"
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Review by jimwang118 — DISMISSED** [review]

**Comment by jimwang118** [comment]
Local test usage found the following issues.
```
Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error at F:/telnetpp/installed/x64-windows/share/telnetpp/telnetpp-targets.cmake:60 (set_target_properties):
  The link interface of target "KazDragon::telnetpp" contains:
    Boost::boost
  but the target was not found.  Possible reasons include:
    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.		F:/telnetpp/installed/x64-windows/share/telnetpp/telnetpp-targets.cmake	60	

Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error in testtelnetpp/CMakeLists.txt:
  Imported target "KazDragon::telnetpp" includes non-existent path
    "F:/telnetpp/installed/x64-windows/include/telnetpp-0.0"
  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
  * The path was deleted, renamed, or moved to another location.
  * An install or uninstall procedure did not complete successfully.
  * The installation package was faulty and references files it does not
  provide.		F:\testwork\testtelnetpp\  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include	
```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by kafeg** [comment]
Sorry, the files `telnetpp-targets.cmake`, `telnetpp-targets-debug.cmake` and `telnetpp-targets-release.cmake` are generated somehow by CMake, not needed when using the lib by vcpkg and I have no idea how to change the genration logic.

So please just decide to merge or decline this PR without this files, I unfortunately can not improve anything else here.
In case of decline, it is ok to me to just use this port as overlay for my needs. Wanted to share new version for community, but that is all what I can propose.

Thank you for review.

File: ports/telnetpp/portfile.cmake [hunk 2]
```diff
@@ -27,8 +26,11 @@ vcpkg_cmake_config_fixup(CONFIG_PATH share/telnetpp)
 
 vcpkg_copy_pdbs()
```
Comment (jimwang118) [inline ports/telnetpp/portfile.cmake:28]:
```suggestion
vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/telnetpp-config.cmake" "####################################################################################" 
                    [[####################################################################################
                      include(CMakeFindDependencyMacro)
                      find_dependency(Boost)
                      find_dependency(gsl-lite)
                      find_dependency(ZLIB)]])
vcpkg_copy_pdbs()

```

File: ports/telnetpp/portfile.cmake [hunk 3]
```diff
@@ -27,8 +26,11 @@ vcpkg_cmake_config_fixup(CONFIG_PATH share/telnetpp)
...
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(REMOVE "${CURRENT_PACKAGES_DIR}/include/telnetpp/version.hpp.in")
-# Handle copyright
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+file(REMOVE 
+    "${CURRENT_PACKAGES_DIR}/include/telnetpp/version.hpp.in" 
+    "${CURRENT_PACKAGES_DIR}/share/telnetpp/telnetpp-targets.cmake" 
+    "${CURRENT_PACKAGES_DIR}/share/telnetpp/telnetpp-targets-debug.cmake" 
+    "${CURRENT_PACKAGES_DIR}/share/telnetpp/telnetpp-targets-release.cmake"
+)
```
Comment (jimwang118) [inline ports/telnetpp/portfile.cmake:35]:
```suggestion
file(REMOVE 
    "${CURRENT_PACKAGES_DIR}/include/telnetpp/version.hpp.in" 
)
```

File: ports/telnetpp/portfile.cmake [hunk 4]
```diff
@@ -1,10 +1,10 @@
 vcpkg_from_github(
   OUT_SOURCE_PATH SOURCE_PATH
   REPO KazDragon/telnetpp
-  REF f370ebd0c0dc6505708065ee5afdc59a6de54387 # v2.1.2 + MSVC patches
-  SHA512 c58cb9159a8fb6c4b089a0212a995f70f08b93877d98828aa263e9f065f42a932d98749b56741d9e711c0805dcc2dcf0607dc86b0553c4e34bd3fad99e0bf157
+  REF "v${VERSION}"
+  SHA512 be0a4304846369f85fef68c9b468b720877a640f8fb32496cf56591da4bb515b9afa9ac4c4477b2275049c304bd17c84b8b82efd8af642c509df452fec9d0d8e
   HEAD_REF master
-  PATCHES fix-install-paths.patch
+  PATCHES fix-install-paths-v3.patch
```
Comment (jimwang118) [inline ports/telnetpp/portfile.cmake:7]:
```suggestion
  PATCHES 
      fix-install-paths-v3.patch
      fix_include.patch
[fix_include.patch](https://github.com/microsoft/vcpkg/files/14327297/fix_include.patch)

```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Review by jimwang118 — APPROVED** [review]

**Comment by jimwang118** [comment]
Usage test pass with following triplets:

```
x64-windows
x64-windows-static
```

**Pull request merged**